.github/GITHUB_WORKFLOW.md

```diff
@@ -0,0 +1,299 @@
+
+To squash your commits, perform an [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History):
+
+1. Check your git branch:
+
+  ```bash
+  git status
+  ```
+
+  The output should be similar to this:
+
+  ```bash
+  On branch your-contribution
+  Your branch is up to date with 'origin/your-contribution'.
+  ```
+
+1. Start an interactive rebase using a specific commit hash, or count backwards from your last commit using `HEAD~<n>`, where `<n>` represents the number of commits to include in the rebase.
+
+  To get this `<n>` value, the easiest way to doing this is running `git log` and then count number of commits until you hit `main` (this may be called `master` instead depending on the repo) but not including `main`.
+
+  ```bash
+  git log
+  ```
+
+  Once you have that number, continue with the rebase and squash.
+
+  ```bash
+  git rebase -i HEAD~3
+  ```
+
+  The output should be similar to this:
+
+  ```bash
+  pick 2ebe926 Original commit
+  pick 31f33e9 Address feedback
+  pick b0315fe Second unit of work
+
+  # Rebase 7c34fc9..b0315ff onto 7c34fc9 (3 commands)
+  #
+  # Commands:
+  # p, pick <commit> = use commit
+  # r, reword <commit> = use commit, but edit the commit message
+  # e, edit <commit> = use commit, but stop for amending
+  # s, squash <commit> = use commit, but meld into previous commit
+  # f, fixup <commit> = like "squash", but discard this commit's log message
+  ...
+  ```
+
+1. Use a command line text editor to change the word `pick` to `squash` for the commits you want to squash, then save your changes and continue the rebase:
+
+  ```bash
+  pick 2ebe926 Original commit
+  squash 31f33e9 Address feedback
+  squash b0315fe Second unit of work
+  ...
+  ```
+
+  The output after saving changes should look similar to this:
+
+  ```bash
+  [detached HEAD 61fdded] Second unit of work
+   Date: Thu Mar 5 19:01:32 2020 +0100
+   2 files changed, 15 insertions(+), 1 deletion(-)
+   ...
+
+  Successfully rebased and updated refs/heads/main.
+  ```
+
+1. Force push your changes to your remote branch:
+
+  ```bash
+  git push --force
+  ```
+
+## 11. Merging a commit
+
+Once you've received review and approval, your commits are squashed, your PR is ready for merging.
+
+Merging happens automatically after both a Reviewer and Approver have approved the PR. If you haven't squashed your commits, they may ask you to do so before approving a PR.
+
+## (If Needed) Reverting a Commit
+
+In case you wish to revert a commit, use the following instructions.
+
+_If you have upstream write access_, please refrain from using the `Revert` button in the GitHub UI for creating the PR, because GitHub will create the PR branch inside the main repository rather than inside your fork.
+
+- Create a branch and sync it with upstream. Note that depending on which repository you are working from, the default branch may be called `master` instead of `main`.
```

Comment:
I think this line isn't necessary since this is specific to the Go SDK

> Note that depending on which repository you are working from, the default branch may be called `master` instead of `main`.